### DISCUSSION TITLE
Performance benchmarking

### DISCUSSION THREAD

**Discussion by ricardobeat** [Starlight-JS/starlight#134] [General]
The README says this project is focused on performance, but I cannot find benchmarks anywhere. Apparently there was an octane bench running in CI but that was removed?

This would be important not only for comparison against other engines, but to measure progress of the project itself, 'arewefastyet' style, making sure improvements have the intended effect and changes don't degrade performance.

**Comment by playXE** [comment]
Hello, @ricardobeat! Yes, there was octane bench in CI but atm Starlight is in a heavy redesign phase on my localhost. When I started this project I had only one thing in mind: "Performance first!" but as it turned out developing everything only to have good performance is not the best way of developing and this made code hard to maintain and lots of UB or bugs could be introduced to code while changing some important parts of runtime. I hope to finish this redesign by the end of this year and publish its code to git. It will feature a custom parser, [Comet](https://github.com/starlight-js/comet) as GC, and unsafe usage should be reduced as much as possible.
